source/framework/core/src/TRestDataSet.cxx

```diff
@@ -321,6 +322,7 @@ std::vector<std::string> TRestDataSet::FileSelection() {
 
         if (!accept) continue;
 
+        fTotalDuration += run->GetEndTimestamp() - run->GetStartTimestamp();
```

Comment:
Maybe it would be good to be able to define the run duration but with a minimum number of events per minute (e.g. 5 or 10 events per minute), so that we don't take into account time in which for example there was a trip and data was not being taken. This way we would not compute the run duration but the effective time of data taking during the run.

Reply:
Right, however, here we are either accepting the file or rejecting the file. This variable registers the total duration from files added to the list.

You could reject the entire file by adding a metadata rule such that the mean rate, captured for example as a metadata via `TRestSummaryProcess`, would exceed certain amount.

However, if rejecting one file is not a solution because the file is the full tracking (for example) and what you want is to partially accept the events inside a given file, then you need to do that later on.

This is because of the nature of adding files inside RDataFrame.

This might need a dedicated method running on the final tree where you find those gaps of data, and reduce the `fTotalDuration` accordingly.

Adding something such as `TRestDataSet::RemoveDataUnderConditions( std::string )`. Probably you need to loop.

I added inside the issue.